File: lana/src/main/scala/com/financialforce/lana/workspace/VSWorkspace.scala
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */
package com.financialforce.lana.workspace

import com.financialforce.lana.Display
import com.financialforce.lana.runtime.vscode.WorkspaceFolder

class VSWorkspace(workspaceFolder: WorkspaceFolder, display: Display) {
  val path: String = workspaceFolder.uri.fsPath
  val name: String = workspaceFolder.name
}

object VSWorkspace {
  def apply(workspaceFolder: WorkspaceFolder, display: Display): VSWorkspace = {
    new VSWorkspace(workspaceFolder, display)
  }
}

File: src/workspace/VSWorkspace.ts
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */

import { WorkspaceFolder } from "vscode";

export class VSWorkspace {
  workspaceFolder: WorkspaceFolder;

  constructor(workspaceFolder: WorkspaceFolder) {
    this.workspaceFolder = workspaceFolder;
  }

  path(): string {
    return this.workspaceFolder.uri.fsPath;
  }
  name(): string {
    return this.workspaceFolder.name;
  }
}
